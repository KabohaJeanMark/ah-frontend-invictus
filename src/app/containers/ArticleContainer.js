import React, { Component } from 'react';
import { connect } from 'react-redux';
import { toast } from 'react-toastify';
import ArticlePresentationComponent from '../components/ArticleComponent';
import { postArticle } from '../../redux/actions/ArticleActions';
import { articleUrl } from '../utils';

export class ArticleContainer extends Component {
  state = {
    title: "",
    description: "",
    body: "",
    tagList: ["tags"]
  };

  checkBlank = () => {
    const { title, body, description } = this.state;
    if (title === '' || body === '' || description === '') {
      toast.dismiss();
      toast.error("All fields must be filled", {
        hideProgressBar: false,
        autoClose: 3000
      });
    }
  };

  handleChange = (event) => {
    this.setState({
      [event.target.name]: event.target.value
    });
  };
  handleSubmit = (event) => {
    event.preventDefault();
    const { title, body, description, tagList } = this.state;
    const articleDetails = { title, body, description, tagList };
    this.checkBlank();
    var headers = {};
    headers['Content-Type'] = 'application/json';
    headers['Authorization'] = `Bearer ${localStorage.getItem('user_token')}`;
    this.props.postArticle(articleUrl, articleDetails, headers);
  };

  render () {
    return (

      <ArticlePresentationComponent
        handleSubmit={this.handleSubmit}
        handleChange={this.handleChange}
      />
    );
  }
}

export default connect(null, { postArticle })(ArticleContainer);
