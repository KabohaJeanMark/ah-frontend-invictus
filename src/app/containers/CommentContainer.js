import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { toast } from 'react-toastify';
import Comment from '../../app/components/Comment';
import { fetchComments } from '../../redux/actions/FetchCommentAction';
import { postComment } from '../../redux/actions/CommentAction';
import FetchComment from '../../app/components/FetchComment';

export class CommentContainer extends Component {
  constructor (props) {
    super(props);

    this.state = {
      commentBody: "",
      displayComment: []
    };
  }
  componentDidMount () {
    const { fetchComments, articleId } = this.props;
    fetchComments(articleId);
  }

  componentWillReceiveProps = (nextProps) => {
    if (nextProps.comments) {
      this.setState({ displayComment: nextProps.comments.comments });
    }
  }

  emptyBody = (comment) => {
    if (comment === undefined) {
      toast.dismiss();
      toast.error("The comment body is required", {
        hideProgressBar: false,
        autoClose: 5000
      });
    }
  };

  handleChange = (event) => {
    this.setState({
      [event.target.name]: event.target.value
    });
  };

  handleSubmit = (event) => {
    const { articleId } = this.props;
    event.preventDefault();
    const { commentBody } = this.state;
    var headers = {};
    headers['Content-Type'] = 'application/json';
    headers['Authorization'] = `Bearer ${localStorage.getItem('user_token')}`;
    this.emptyBody(commentBody);
    this.props.postComment(articleId, commentBody);
  };

  render () {
    const { comments } = this.props;
    return (

      <React.Fragment>
        <Comment
          handleSubmit={this.handleSubmit}
          handleChange={this.handleChange}
        />
        <FetchComment comments={comments} />
      </React.Fragment>
    );
  }
}

export const mapStateToProps = state => ({
  comment: state.postComment,
  comments: state.fetchComment.comments
});

CommentContainer.propTypes = {
  fetchComment: PropTypes.func
};

export default connect(mapStateToProps, { postComment, fetchComments })(CommentContainer);
