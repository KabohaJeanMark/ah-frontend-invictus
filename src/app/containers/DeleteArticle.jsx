import React, { Component } from "react";
import { connect } from "react-redux";
import { deleteArticleAction } from '../../redux/actions/DeleteArticle';
import DeleteArticleComponent from '../components/DeleteArticle';

export class DeleteArticleContainer extends Component {
  handleDelete = event => {
    /* eslint-disable no-alert*/
    const delArticle = window.confirm("Are you sure you want to delete this article");
    if (delArticle === true) {
      this.props.deleteArticleAction(this.props.id);
    }
  }
  render () {
    return (
      <>
        < DeleteArticleComponent
          handleDelete={this.handleDelete}
        />
      </>
    );
  }
}

export const mapStateToProps = state => ({
  deleteArticleErrors: state.deleteArticle.deleteArticleErrors,
  deleteArticleResponse: state.deleteArticle.deleteArticleResponse
});

export default connect(
  mapStateToProps,
  { deleteArticleAction }
)(DeleteArticleContainer);
