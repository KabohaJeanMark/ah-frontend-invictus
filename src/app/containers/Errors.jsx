import React, { Component } from 'react';

export default class Errors extends Component {
  constructor (props) {
    super(props);
    this.state = { errors: [] };
  }
  componentWillReceiveProps (nextProps) {
    if (nextProps.errors) {
      this.setState({
        errors: this.mapErrorsToList(nextProps)
      });
    }
  }
   mapErrorsToList = props => {
     let items = [];
     if (Array.isArray(props.errors)) {
       for (let i = 0; i < props.errors.length; i++) {
         items.push(props.errors[i]);
       }
     } else {
       Object.keys(props.errors).forEach(key => {
         for (let i = 0; i < props.errors[key].length; i++) {
           items.push(key + ': ' + props.errors[key][i]);
         }
       });
     }
     return items;
   }
   render () {
     function Repeat (props) {
       let items = [];
       for (let i = 0; i < props.errors.length; i++) {
         items.push(props.children(i, props.errors[i]));
       }
       return <div className="danger" ><ul>{items}</ul></div>;
     }
     return (
       this.state.errors.length ? (
         <Repeat errors={this.state.errors}>
           {(index, message) => <li key={index}>{message}</li>}
         </Repeat>
       ) : (null)
     );
   }
}
