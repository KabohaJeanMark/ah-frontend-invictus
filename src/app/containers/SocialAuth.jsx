import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import FacebookSocialAuth from 'react-facebook-login';
import GoogleSocialAuth from 'react-google-login';
import socialAuth from '../../redux/actions/socialAuth/socialAuthAction';
import '../../assets/styles/socialbuttons.scss';

export class SocialAuthentication extends Component {
  responseFacebook = (response) => {
    const { socialAuth } = this.props;
    socialAuth(response.accessToken, 'facebook');
  };

  googleResponse = (response) => {
    const { socialAuth } = this.props;
    const accessToken = socialAuth(response.tokenId, 'google');
    return accessToken;
  };

  render () {
    return (
      <div className="socialAuth">
        <FacebookSocialAuth
          appId="349961962392067"
          fields="name,email"
          callback={this.responseFacebook}
          textButton={false}
          cssClass="fa fa-facebook"
        />
        <GoogleSocialAuth
          clientId="45687926143-vk1re0t33ljt0f46pr63gifaluq8q0sn.apps.googleusercontent.com"
          render={renderProps => (
            <button className="fa fa-google" onClick={renderProps.onClick} disabled={renderProps.disabled}>
            </button>
          )}
          onSuccess={this.googleResponse}
          onFailure={this.googleResponse}
        />
      </div>
    );
  }
}

SocialAuthentication.propType = {
  responseFacebook: PropTypes.func,
  responseGoogle: PropTypes.func
};

export const mapStateToProps = state => ({
  socialAuthState: state.socialAuthReducer
});

export default connect(mapStateToProps, { socialAuth })(SocialAuthentication);
