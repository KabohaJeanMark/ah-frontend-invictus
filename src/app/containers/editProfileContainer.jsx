import React, { Component } from 'react';
import { connect } from 'react-redux';
import firebase from '../../firebase/config';
import { editProfile } from '../../redux/actions/profileActions';
import '../../assets/styles/userprofile.scss';
import NavbarComponent from '../components/Navbar';

export class EditProfileContainer extends Component {
  constructor (props) {
    super(props);
    this.state = {
      firstname: "",
      lastname: "",
      bio: "",
      image: "",
      progress: 0,
      uploadDone: false
    };
  }

  componentWillMount () {
    const { firstname, lastname, bio, image } = this.props.profile.profile;
    this.setState({ firstname, lastname, bio, image });
  }

  componentWillReceiveProps (nextProps) {
    if (nextProps.editSuccess === true) {
      const { history } = nextProps;
      history.push('/profile');
    }
  }

  onChange = (e) => {
    this.setState({ [e.target.name]: e.target.value });
  }
  onSubmit = (e) => {
    e.preventDefault();
    const { firstname, lastname, bio, image } = this.state;
    const newProfile = {
      firstname: firstname,
      lastname: lastname,
      bio: bio,
      image: image
    };
    this.props.editProfile(newProfile, this.props);
  }

  /* istanbul ignore next */
  uploadImage (files) {
    const fileload = firebase
      .storage()
      .ref(`images/${files[0].name}`)
      .put(files[0]);

    fileload.then(res => {
      firebase
        .storage()
        .ref(`images/${files[0].name}`)
        .getDownloadURL()
        .then(url => {
          const image = {
            image: url
          };
          this.props.editProfile(image);
        });
    });
    fileload.on("state_changed", snapshot => {
      const uploadDone = true;
      this.setState({ uploadDone });
      const progress = Math.round(
        (snapshot.bytesTransferred / snapshot.totalBytes) * 100
      );
      this.setState({ progress });
    });
  }

  render () {
    return (
      <div>
        <div className="row">
          <div className="col-md-12"><NavbarComponent /></div>
        </div>
        <div className="row">
          <div className="col-md-4" />
          <div className="col-md-4">
            <div className="image-div" >
              {this.state.image === "" ? (<img alt="profileimage" className="profile-image"
                src={require("../../assets/images/avatar2.png")} />) :
                (<img className="profile-image" alt="profileimage" src={this.state.image} />)}
              <progress value={this.state.progress} max="100" />
              <input
                type="file"
                onChange={event => this.uploadImage(event.target.files)}
                className="upload-button mt-3"
                id="profile"
              />
            </div>
            <form onSubmit={this.onSubmit}>
              <div>
                <div className="form-group">
                  <label className="">firstname</label>
                  <input type="text" name="firstname"
                    onChange={this.onChange} className="form-control"
                    defaultValue={this.state.firstname} />
                </div>
                <div className="form-group">
                  <label className="">Lastname</label>
                  <input type="text" name="lastname" className="form-control"
                    onChange={this.onChange}
                    defaultValue={this.state.lastname} />
                </div>
                <div className="form-group">
                  <label className="">Bio</label>
                  <textarea className="form-control" name="bio" id="exampleFormControlTextarea1"
                    rows="3" onChange={this.onChange} value={this.state.bio} />
                </div>
              </div>
              <div><button className="button" type="submit">Save Changes</button></div>
            </form>
          </div>
          <div className="col-md-4" />
        </div>
      </div>
    );
  }
}


export const mapStateToProps = state => ({
  profile: state.profileReducer,
  editSuccess: state.profileReducer.editSuccess
});

export default connect(mapStateToProps, { editProfile })(EditProfileContainer);
