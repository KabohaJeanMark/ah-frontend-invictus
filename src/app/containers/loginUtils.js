export const loginUtil = {
  LOGIN_UTILS: /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/
};
