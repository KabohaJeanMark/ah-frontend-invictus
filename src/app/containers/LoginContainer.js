import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { loginAction } from '../../redux/actions/LoginActions';
import Login from '../components/Login';
import { loginUtil } from './loginUtils';

export class LoginContainer extends Component {
  constructor () {
    super();
    this.state = {
      email: '',
      password: '',
      errors: {}
    };
  }
  onChange = (event) => {
    event.preventDefault();
    this.setState({ [event.target.name]: event.target.value });
  }
  checkValidEmail = (userEmail) => {
    const mailFormat = loginUtil.LOGIN_UTILS;
    if (userEmail.match(mailFormat)) {
      return true;
    }
    return false;
  };

  checkPassword = (userPassword) => {
    if (userPassword === undefined) {
      return false;
    }
    return true;
  };

  handleSubmit = (event) => {
    event.preventDefault();
    const userData = {
      user: {
        email: this.state.email,
        password: this.state.password
      }
    };
    if (this.state.email === undefined) {
      this.setState({ errors: { userEmail: 'Email required' } });
    } else if (!this.checkValidEmail(this.state.email)) {
      this.setState({ errors: { userEmail: 'invalid email format' } });
    } else if (!this.checkPassword(this.state.password)) {
      this.setState({ errors: { userPassword: 'Password field is empty' } });
    } else {
      this.props.loginAction(userData, this.props);
    }
  }
  render () {
    return (
      <Login
        handleSubmit={this.handleSubmit}
        onChange={this.onChange}
        email={this.state.email}
        password={this.state.password}
        errors={this.state.errors}
      />
    );
  }
}

LoginContainer.propTypes = {
  email: PropTypes.string,
  password: PropTypes.string,
  handleSubmit: PropTypes.func,
  onChange: PropTypes.func,
  loginAction: PropTypes.func,
  errors: PropTypes.object
};

LoginContainer.defaultProps = {
  errors: {},
  email: '',
  password: '',
  handleSubmit: () => { },
  onChange: () => { }
};
export const mapStateToProps = state => ({ login: state.auth });
export default connect(mapStateToProps, { loginAction })(LoginContainer);
