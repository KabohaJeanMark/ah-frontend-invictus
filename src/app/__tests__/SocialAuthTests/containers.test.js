import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import { SocialAuthentication, mapStateToProps } from '../../containers/SocialAuth';
import socialAuthReducer from '../../../redux/reducers/socialAuth';
import { socialTypes } from '../../../redux/actions/types';
import socialAuth from '../../../redux/actions/socialAuth/socialAuthAction';

const storeItems = {};
const mockStore = configureStore();
const store = mockStore(storeItems);

const initialState = {
  isAuthenticated: false,
  facebookLogin: false,
  payload: '',
  token: ''
};

const initialStateGoogle = {
  isAuthenticated: false,
  googleLogin: false,
  payload: '',
  token: ''
};

const props = {
  socialAuth: jest.fn(),
  onClick: jest.fn()
};

const setUp = () => {
  const social = shallow(<SocialAuthentication store={store} {...props} />);
  return social;
};

describe('Social Auth Reducer', () => {
  it('tests inititial state of social auth reducer', () => {
    const newState = socialAuthReducer(initialState, {});
    expect(newState).toEqual(initialState);
  });

  it('tests successful authentication through facebook', () => {
    let newState;
    newState = socialAuthReducer(initialState, {
      type: socialTypes.FACEBOOK_AUTH,
      payload: {
        email: '',
        message: '',
        token: '',
        username: ''
      },
      token: ''
    });
    expect(newState).toEqual(
      {
        isAuthenticated: true,
        facebookLogin: true,
        payload: {
          email: '',
          message: '',
          token: '',
          username: '' },
        token: ''
      }
    );
  });

  it('tests successful authentication through google', () => {
    let newState;
    newState = socialAuthReducer(initialStateGoogle, {
      type: socialTypes.GOOGLE_AUTH,
      payload: {
        email: '',
        message: '',
        token: '',
        username: ''
      },
      token: ''
    });
    expect(newState).toEqual(
      {
        isAuthenticated: true,
        googleLogin: true,
        payload: {
          email: '',
          message: '',
          token: '',
          username: '' },
        token: ''
      }
    );
  });
});

describe('facebook social authentication', () => {
  it('matches snapshot', () => {
    const wrapper = setUp();
    mapStateToProps(initialState);
    wrapper.instance().responseFacebook(socialAuth());
    expect(wrapper.instance()).toMatchSnapshot();
  });
});

describe('google social authentication', () => {
  it('matches snapshot', () => {
    const wrapper = setUp();
    wrapper.instance().googleResponse(socialAuth());
    expect(wrapper.instance()).toMatchSnapshot();
  });
});
