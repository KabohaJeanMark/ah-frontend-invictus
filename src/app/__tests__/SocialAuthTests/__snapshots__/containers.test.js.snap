// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`facebook social authentication matches snapshot 1`] = `
SocialAuthentication {
  "context": Object {},
  "googleResponse": [Function],
  "props": Object {
    "onClick": [MockFunction],
    "socialAuth": [MockFunction] {
      "calls": Array [
        Array [
          undefined,
          "facebook",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    },
    "store": Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    },
  },
  "refs": Object {},
  "responseFacebook": [Function],
  "setState": [Function],
  "state": null,
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useState": [Function],
      },
      "_element": <SocialAuthentication
        onClick={[MockFunction]}
        socialAuth={
          [MockFunction] {
            "calls": Array [
              Array [
                undefined,
                "facebook",
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
        store={
          Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
        }
      />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <div
        className="socialAuth"
      >
        <t
          appId="349961962392067"
          callback={[Function]}
          cssClass="fa fa-facebook"
          fields="name,email"
          size="metro"
          tag="button"
          textButton={false}
          typeButton="button"
        />
        <e
          accessType="online"
          buttonText="Sign in with Google"
          clientId="45687926143-vk1re0t33ljt0f46pr63gifaluq8q0sn.apps.googleusercontent.com"
          cookiePolicy="single_host_origin"
          disabledStyle={
            Object {
              "opacity": 0.6,
            }
          }
          fetchBasicProfile={true}
          icon={true}
          isSignedIn={false}
          jsSrc="https://apis.google.com/js/api.js"
          onFailure={[Function]}
          onRequest={[Function]}
          onSuccess={[Function]}
          prompt=""
          render={[Function]}
          scope="profile email"
          tag="button"
          theme="light"
          type="button"
          uxMode="popup"
        />
      </div>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
}
`;

exports[`google social authentication matches snapshot 1`] = `
SocialAuthentication {
  "context": Object {},
  "googleResponse": [Function],
  "props": Object {
    "onClick": [MockFunction],
    "socialAuth": [MockFunction] {
      "calls": Array [
        Array [
          undefined,
          "facebook",
        ],
        Array [
          undefined,
          "google",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    },
    "store": Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    },
  },
  "refs": Object {},
  "responseFacebook": [Function],
  "setState": [Function],
  "state": null,
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useState": [Function],
      },
      "_element": <SocialAuthentication
        onClick={[MockFunction]}
        socialAuth={
          [MockFunction] {
            "calls": Array [
              Array [
                undefined,
                "facebook",
              ],
              Array [
                undefined,
                "google",
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
        store={
          Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
        }
      />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <div
        className="socialAuth"
      >
        <t
          appId="349961962392067"
          callback={[Function]}
          cssClass="fa fa-facebook"
          fields="name,email"
          size="metro"
          tag="button"
          textButton={false}
          typeButton="button"
        />
        <e
          accessType="online"
          buttonText="Sign in with Google"
          clientId="45687926143-vk1re0t33ljt0f46pr63gifaluq8q0sn.apps.googleusercontent.com"
          cookiePolicy="single_host_origin"
          disabledStyle={
            Object {
              "opacity": 0.6,
            }
          }
          fetchBasicProfile={true}
          icon={true}
          isSignedIn={false}
          jsSrc="https://apis.google.com/js/api.js"
          onFailure={[Function]}
          onRequest={[Function]}
          onSuccess={[Function]}
          prompt=""
          render={[Function]}
          scope="profile email"
          tag="button"
          theme="light"
          type="button"
          uxMode="popup"
        />
      </div>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
}
`;
